File: backend/app/agents/writer_agent.py
from langchain_core.prompts import ChatPromptTemplate
from langchain_core.output_parsers import StrOutputParser
from app.core.llm import get_llm

llm = get_llm()

writer_prompt = ChatPromptTemplate.from_messages([
    ("system", "You are an expert research writer."),
    ("human", """Write a detailed research report.

Topic: {topic}

Research:
{research}

Structure:
- Introduction
- Key Points
- Conclusion
- Sources
"""),
])

writer_chain = writer_prompt | llm | StrOutputParser()

File: syncai/src/lib/agents/writerAgent.ts
import { getModel } from "../llm";

export async function runWriterAgent(
  topic: string,
  research: string,
): Promise<string> {
  const model = getModel();
  const result = await model.generateContent(
    `You are an expert research writer. Write a detailed, well-structured research report in Markdown format.

Topic: ${topic}

Research Material:
${research.slice(0, 3000)}

Write a comprehensive report with these sections:
# ${topic}

## Introduction
## Key Findings
## Analysis
## Conclusion
## Sources

Use Markdown formatting including headers, bullet points, and bold text where appropriate:`,
  );

  return result.response.text();
}
